### PR TITLE
HEEDLS-865 Add endpoint with json centres data for find your centre map

### PR DESCRIPTION
**Pull request by AlexJacksonDS**
### JIRA link
https://softwiretech.atlassian.net/browse/HEEDLS-865

### Description
Added an endpoint that returns JSON data for centres to display on the Find Your Centre map

### Screenshots
N/A
-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors.
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript. Full manual testing guidelines can be found here: https://softwiretech.atlassian.net/wiki/spaces/HEE/pages/6703648740/Testing
- [ ] Updated/added documentation in Swiki and/or Readme. Links (if any) are below:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken.
- [x] Scanned over my own MR to ensure everything is as expected.

### CODE REVIEW

File: DigitalLearningSolutions.Data.Tests/Services/CentresServiceTests.cs [hunk 1]
```diff
@@ -87,5 +87,21 @@ public void GetAllCentreSummariesForSuperAdmin_calls_dataService_and_returns_all
...
+            var centres = Builder<CentreSummaryForMap>.CreateListOfSize(10).Build();
+            A.CallTo(() => centresDataService.GetAllCentreSummariesForMap()).Returns(centres);
+
+            // When
+            var result = centresService.GetAllCentreSummariesForMap();
+
+            // Then
+            result
+                .Should()
+                .HaveCount(10);
```
Comment (livzorn) [inline DigitalLearningSolutions.Data.Tests/Services/CentresServiceTests.cs:104]: not important but this looks weirdly auto-formatted?

**Review by livzorn — APPROVED** [review]
Nice work! Just left one small note

File: DigitalLearningSolutions.Web/Controllers/FindYourCentreController.cs [hunk 2]
```diff
@@ -27,5 +30,12 @@ public IActionResult Index(string? centreId)
 
             return View(model);
         }
+
+        [RedirectDelegateOnlyToLearningPortal]
```
Comment (DanBloxham-sw) [inline DigitalLearningSolutions.Web/Controllers/FindYourCentreController.cs:34]: Can we apply this to the controller instead?

File: DigitalLearningSolutions.Data.Tests/Services/CentresServiceTests.cs (hunk 1 above)
Comment (DanBloxham-sw) [reply DigitalLearningSolutions.Data.Tests/Services/CentresServiceTests.cs:104]: Also do we want to check the result is equivalent to the centres?

**Review by DanBloxham-sw — CHANGES_REQUESTED** [review]
Couple of minor points plus Olivia's comments, but otherwise looks good.

**Review by DanBloxham-sw — APPROVED** [review]
Looks good.

**Pull request merged**